author: Andres Salomon <dilinger@debian.org>

This is some clang-18 stuff. For clang-16, the compiler chokes on these
flags. These used to only be enabled when llvm_force_head_revision=true,
but now they're enabled unconditionally.

--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -573,15 +573,10 @@ config("compiler") {
     # TODO(crbug.com/40284925): This isn't supported by Cronet's mainline llvm version.
     if (default_toolchain != "//build/toolchain/cros:target" &&
         !llvm_android_mainline) {
-      cflags += [
-        "-mllvm",
-        "-split-threshold-for-reg-with-hint=0",
-      ]
       if (use_thin_lto && is_a_target_toolchain) {
         if (is_win) {
           ldflags += [ "-mllvm:-split-threshold-for-reg-with-hint=0" ]
         } else {
-          ldflags += [ "-Wl,-mllvm,-split-threshold-for-reg-with-hint=0" ]
         }
       }
     }
@@ -1891,7 +1886,6 @@ config("default_warnings") {
         "-Wno-vla-extension",
 
         # TODO(crbug.com/40284799): Fix and re-enable.
-        "-Wno-thread-safety-reference-return",
       ]
 
       cflags_cc += [
@@ -1902,7 +1896,7 @@ config("default_warnings") {
       if (!is_nacl) {
         cflags_cc += [
           # TODO(crbug.com/41486292): Fix and re-enable.
-          "-Wno-c++11-narrowing-const-reference",
+          "-Wno-c++11-narrowing",
         ]
       }
     }
